pkg/dns/ibm/common.go

```diff
@@ -4,18 +4,44 @@ import (
 	"fmt"
 	"regexp"
 
+	"github.com/IBM/networking-go-sdk/dnsrecordsv1"
+	"github.com/IBM/networking-go-sdk/dnssvcsv1"
+
 	configv1 "github.com/openshift/api/config/v1"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 
 	iov1 "github.com/openshift/api/operatoringress/v1"
 )
 
+const (
+	// CISCustomEndpointName is the key used to identify the CIS service in ServiceEndpoints
+	CISCustomEndpointName = "cis"
+	// DNSCustomEndpointName is the key used to identify the DNS service in ServiceEndpoints
+	DNSCustomEndpointName = "dns"
+	// IAMCustomEndpointName is the key used to identify the IAM service in ServiceEndpoints
+	IAMCustomEndpointName = "iam"
+	// default iam server url
+	defaultIamTokenServerEndpoint = "https://iam.cloud.ibm.com"
+)
+
 var (
 	// IBMResourceCRNRegexp is a regular expression of IBM resource CRNs
 	// See: https://cloud.ibm.com/docs/account?topic=account-crn
 	IBMResourceCRNRegexp = regexp.MustCompile(`^crn:v[0-9]:(?P<cloudName>[^:]*):(?P<cloudType>[^:]*):(?P<serviceName>[^:]*):(?P<location>[^:]*):(?P<scope>[^:]*):(?P<guid>[^:]*):(?P<resourceType>[^:]*):(?P<resourceID>[^:]*)$`)
 )
 
+// ServiceEndpoint stores the configuration of a custom url to
+// override default IBM Service API endpoints.
+type ServiceEndpoint struct {
```

Comment:
Shouldn't we just use `map<serviceName,URL>` instead? By default it would contain the default URLs, which would be overwritten by custom ones if there are any configured.

Reply:
I thought to keep it similar to https://github.com/Karthik-K-N/cluster-ingress-operator/blob/be08afc7f875d98d48faf17e88c015451082c00b/vendor/github.com/openshift/api/config/v1/types_infrastructure.go#L709-L731

Reply:
I see that your implementation is consistent with the aws serviceEndpoints implementation, so I'll let you and the code owners decide.